Assets/MixedRealityToolkit.SDK/Features/Utilities/Solvers/Solver.cs

```diff
@@ -210,12 +210,18 @@ protected virtual void Awake()
         }
 
         /// <summary>
-        /// Typically when a solver becomes enabled, it should update its internal state to the system, in case it was disabled far away
+        /// When a solver becomes enabled, it must make sure it is registered at its <see cref="SolverHandler"/>.
+        /// Also it should update its internal state to the system, in case it was disabled far away.
         /// </summary>
         protected virtual void OnEnable()
         {
             if (SolverHandler != null)
             {
+                if(!SolverHandler.Solvers.Contains(this))
+                { 
+                    SolverHandler.Solvers = SolverHandler.GetComponents<Solver>();
```

Comment:
SO I can see this is the crux of the change, however is multiple solvers get added / removed, this is going to be refreshed / updated multiple times.  Might be better to have a call back to the SolverHandler to refresh it's solvers and check the "Solvers" list against the component list, instead of repeatedly calling "GetComponents" which is costly.

Reply:
Had same comment, though in more words, maybe let's have this chat in my slightly longer comment.